pyqtgraph/SignalProxy.py

```diff
@@ -26,86 +29,81 @@ def __init__(self, signal, delay=0.3, rateLimit=0, slot=None):
         rateLimit - (signals/sec) if greater than 0, this allows signals to stream out at a 
                     steady rate while they are being received.
         """
-        
+
         QtCore.QObject.__init__(self)
-        signal.connect(self.signalReceived)
-        self.signal = signal
         self.delay = delay
         self.rateLimit = rateLimit
         self.args = None
         self.timer = ThreadsafeTimer.ThreadsafeTimer()
         self.timer.timeout.connect(self.flush)
-        self.blockSignal = False
-        self.slot = weakref.ref(slot)
         self.lastFlushTime = None
+        self.signal = signal
+        self.signal.connect(self.signalReceived)
         if slot is not None:
+            self.blockSignal = False
             self.sigDelayed.connect(slot)
-        
+            self.slot = weakref.ref(slot)
+        else:
+            self.blockSignal = True
+            self.slot = None
+
     def setDelay(self, delay):
         self.delay = delay
-        
+
     def signalReceived(self, *args):
-        """Received signal. Cancel previous timer and store args to be forwarded later."""
+        """Received signal. Cancel previous timer and store args to be
+        forwarded later."""
         if self.blockSignal:
             return
         self.args = args
         if self.rateLimit == 0:
             self.timer.stop()
-            self.timer.start((self.delay*1000)+1)
+            self.timer.start((self.delay * 1000) + 1)
```

Comment:
`timer.start()` needs to take an `int` or else it spews a deprecation warning; can you have the starting parameter be `int(self.delay * 1000) + ``?